### PR TITLE
[SYCL][NFC] Detach unit-tests naming from PI

### PR DESCRIPTION
**Pull request by AlexeySachkov**
This is a by-product of #14145. Our unit tests have mock classes corresponding to device images and other entries our compiler generates. They are all named using `Pi` prefix, but they have nothing to do with Plugin Interface, they are part of our compiler interface.

This patch renames associated files and classes so they do not refer to `Pi` anymore.
This is a mechanical find-and-replace change.

### CODE REVIEW

**Review by bso-intel — APPROVED** [review]

File: sycl/unittests/SYCL2020/DeviceGetInfoAspects.cpp [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 #include <sycl/sycl.hpp>
 
-#include <helpers/PiImage.hpp>
+#include <helpers/MockDeviceImage.hpp>
```
Comment (bader) [inline sycl/unittests/SYCL2020/DeviceGetInfoAspects.cpp:11]: Does the test need this include?

File: sycl/unittests/accessor/AccessorPlaceholder.cpp [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #include <gtest/gtest.h>
 
-#include <helpers/PiImage.hpp>
+#include <helpers/MockDeviceImage.hpp>
```
Comment (bader) [inline sycl/unittests/accessor/AccessorPlaceholder.cpp:3]: Does the test need this include?

File: sycl/unittests/SYCL2020/DeviceGetInfoAspects.cpp (hunk 1 above)
Comment (AlexeySachkov) [reply sycl/unittests/SYCL2020/DeviceGetInfoAspects.cpp]: Removed the include in 3aa939f4dd81b176fedb6303d41d7599b4acdced

File: sycl/unittests/accessor/AccessorPlaceholder.cpp (hunk 2 above)
Comment (AlexeySachkov) [reply sycl/unittests/accessor/AccessorPlaceholder.cpp]: Removed the include in 3aa939f4dd81b176fedb6303d41d7599b4acdced

**Comment by AlexeySachkov** [comment]
Superseded by #14815

**Pull request closed without merging**